control-plane/api-gateway/cache/gateway.go

```diff
@@ -125,7 +125,13 @@ func (r *GatewayCache) subscribeToGateway(ctx context.Context, ref api.ResourceR
 
 			return nil
 		}, backoff.WithContext(retryBackoff, ctx)); err != nil {
-			r.logger.Error(err, fmt.Sprintf("unable to fetch config entry for gateway: %s/%s", ref.Namespace, ref.Name))
+			// if we timeout we don't care about the error message because it's expected to happen on long polls
+			// any other error we want to alert on
+			if !strings.Contains(strings.ToLower(err.Error()), "timeout") &&
+				!strings.Contains(strings.ToLower(err.Error()), "no such host") &&
+				!strings.Contains(strings.ToLower(err.Error()), "connection refused") {	
+				r.logger.Error(err, fmt.Sprintf("unable to fetch config entry for gateway: %s/%s", ref.Namespace, ref.Name))
+			}
```

Comment:
I wish we had a better way to do error handling in general, but this matches what we're doing elsewhere
https://github.com/hashicorp/consul-k8s/blob/2d03f3e7c513f3abde003040ebdbc53a8b30858b/control-plane/api-gateway/cache/consul.go#L196-L206